db_access.py

```diff
@@ -27,6 +27,27 @@ def get_all_users():
     return conn.execute('SELECT id, handle, admin FROM user').fetchall()
 
 
+def user_for(uid):
+    conn = get_db()
+    for a in conn.execute('SELECT handle FROM user WHERE id=:id', {'id':uid}):
+        return a[0]
+    return ''
+
+
+def getUserLike(like_handle):
+    conn = get_db()
+    like_handle = '%' + like_handle + '%'
+    return list(conn.execute('SELECT * FROM user WHERE handle LIKE :like_handle', {'like_handle':like_handle}))
+   
+    
+def sign_in(handle, password):
+    conn = get_db()
+    password = hash_ps(password)
+    for c in conn.execute('SELECT id FROM user WHERE handle=:handle AND password=:password', {'handle':handle, 'password':password}):
+        return (True, c[0])
+    return (False, -1)
```

Comment:
This method is confusing. It is called `sign_in` but it does not actually sign the user it. What it does do is return a tuple that contains information about whether the user _can_ be signed it.

Here in the `db_access` I think the functions should be as close to simple DB calls as is reasonable. So I would call this method `get_user_by_handle_and_password` and it would return a user or `None`

You can again use `fetchone()` to simplify. A quick check to the [docs](https://docs.python.org/3/library/sqlite3.html#sqlite3.Cursor.fetchone) shows that `fetchone` returns `None` when nothing is available.